tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -4146,7 +4162,566 @@ TEST_F(ExecutionTest, ATOSampleCmpLevelTest) {
+
+  D3D12_FILTER filters[] = {D3D12_FILTER_MIN_MAG_LINEAR_MIP_POINT,
+                            D3D12_FILTER_COMPARISON_MIN_MAG_LINEAR_MIP_POINT};
+  CreateDefaultSamplers(pDevice, pSampHeap->GetCPUDescriptorHandleForHeapStart(),
+                        filters, nullptr /*borderColors*/, 1);
+
+  // Set Heaps, Rootsignature and table
+  ID3D12DescriptorHeap *const pHeaps[2] = { pResHeap, pSampHeap };
+  pCommandList->SetDescriptorHeaps(2, pHeaps);
+  pCommandList->SetComputeRootSignature(pRootSignature);
+  pCommandList->SetComputeRootDescriptorTable(0, pResHeap->GetGPUDescriptorHandleForHeapStart());
+  pCommandList->SetComputeRootDescriptorTable(1, pSampHeap->GetGPUDescriptorHandleForHeapStart());
+
+  // dispatch and close shader
+  pCommandList->Dispatch(1, 1, 1);
+
+  // Copy the results back to readable memory
+  CD3DX12_RESOURCE_BARRIER barrier = CD3DX12_RESOURCE_BARRIER::Transition(pOutputResource,
+                                        D3D12_RESOURCE_STATE_UNORDERED_ACCESS, D3D12_RESOURCE_STATE_COPY_SOURCE);
+  pCommandList->ResourceBarrier(1, &barrier);
+  pCommandList->CopyResource(pOutputReadBuffer, pOutputResource);
+
+  pCommandList->Close();
+
+  pCommandQueue->ExecuteCommandLists(1, ppCommandLists);
+  WaitForSignal(pCommandQueue, FO);
+
+  MappedData mappedData(pOutputReadBuffer, 4*valueSizeInBytes);
+  GatherType *pData = (GatherType*)mappedData.data();
+  GatherType *texVals = (GatherType*)rawTex->GetElements();
+  UINT yCt = yDim;
+  UINT xCt = xDim;
+#ifdef RAWGATHER_FALLBACK
+  // 64-bit fallback uses Load, which doesn't support clamp addressing. so don't test it
+  if (sizeof(GatherType) == 8) {
+    yCt--;
+    xCt--;
+  }
+#endif
+  for (UINT y = 0; y < yCt; y++) {
+    UINT yp1 = y+1>=yDim?y:y+1;
+    for (UINT x = 0; x < xCt; x++) {
+      UINT xp1 = x+1>=xDim?x:x+1;
+      // Because this order may be unexpected, I'll quote the spec:
+      // "The four samples that would contribute to filtering are placed into xyzw
+      //  in counter clockwise order starting with the sample to the lower left"
+      VERIFY_ARE_EQUAL(pData[4*(32*y + x)+0], texVals[yp1*xDim + x]);
+      VERIFY_ARE_EQUAL(pData[4*(32*y + x)+1], texVals[yp1*xDim + xp1]);
+      VERIFY_ARE_EQUAL(pData[4*(32*y + x)+2], texVals[y*xDim   + xp1]);
+      VERIFY_ARE_EQUAL(pData[4*(32*y + x)+3], texVals[y*xDim   + x]);
+    }
+  }
+}
+
+// Create textures of various types and alias them to the unsigned integer format
+// that has the same element size and initializes them with various values,
+// The shader code copies the results of raw gather to an unsigned integer UAV
+// The UAV contents are compared to the values assigned to the texture
+TEST_F(ExecutionTest, ATORawGather) {
+
+  WEX::TestExecution::SetVerifyOutput verifySettings(WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  CComPtr<ID3D12Device> pDevice;
+#ifdef RAWGATHER_FALLBACK
+  D3D_SHADER_MODEL sm = D3D_SHADER_MODEL_6_6;
```

Comment:
Is it possible to make fallback work on an earlier shader model?  What SM 6.6 feature does the fallback path require?

Reply:
I was successfully able to run the fallback with SM 6.2 (when adding `-enable-16bit-types` option).